pkg/dependencies/dependents.go

```diff
@@ -0,0 +1,237 @@
+//   - A map where the key is a string representing the package name, and the value is a dependencyNode containing
+//     the dependents of the package.
+//   - An error
+func findDependentsOfDependencies(ctx context.Context, gqlClient graphql.Client) (map[string]dependencyNode, error) {
+	// Initialize a map to hold package names mapped to their dependency nodes. This will be the returned map
+	packages := make(map[string]dependencyNode)
+
+	// Initialize maps to hold dependency and dependent edges.
+	// dependentEdges maps a package ID to the IDs of packages that depend on it (DepPkg to Pkg).
+	// dependencyEdges maps a package ID to the IDs of its dependencies (Pkg to DepPkg).
+	dependentEdges := make(map[string][]string)  // id -> dependents ids
+	dependencyEdges := make(map[string][]string) // id -> dependency ids
+
+	// Initialize a map to convert package IDs to their names.
+	idToName := make(map[string]string)
+
+	sboms, err := model.HasSBOMs(ctx, gqlClient, model.HasSBOMSpec{})
+
+	if err != nil {
+		return nil, fmt.Errorf("error getting dependencies: %v", err)
+	}
+
+	for _, resp := range sboms.HasSBOM {
+		// Skip entries from "deps.dev" because they are inconsistent.
+		if resp.Origin == deps_dev.DepsCollector {
+			continue
+		}
+		// Iterate through the included dependencies of each SBOM.
+		for _, dependency := range resp.IncludedDependencies {
+			// Construct unique names for the dependency package and the package itself.
+			// TODO: Make the names actually unique, not just add "_".
+			depPkgName := dependency.DependencyPackage.Type + "_" + dependency.DependencyPackage.Namespaces[0].Namespace + "_" + dependency.DependencyPackage.Namespaces[0].Names[0].Name
+			pkgName := dependency.Package.Type + "_" + dependency.Package.Namespaces[0].Namespace + "_" + dependency.Package.Namespaces[0].Names[0].Name
+
+			var depPkgIds []string
+			pkgId := dependency.Package.Namespaces[0].Names[0].Versions[0].Id
+
+			if len(dependency.DependencyPackage.Namespaces[0].Names[0].Versions) == 0 {
+				findMatchingDepPkgVersionIDs, err := findDepPkgVersionIDs(ctx, gqlClient, dependency.DependencyPackage.Type,
+					dependency.DependencyPackage.Namespaces[0].Namespace,
+					dependency.DependencyPackage.Namespaces[0].Names[0].Name, dependency.VersionRange)
+				if err != nil {
+					return nil, fmt.Errorf("error from FindMatchingDepPkgVersionIDs:%w", err)
+				}
+				depPkgIds = append(depPkgIds, findMatchingDepPkgVersionIDs...)
+			} else {
+				depPkgIds = append(depPkgIds, dependency.DependencyPackage.Namespaces[0].Names[0].Versions[0].Id)
+			}
+
+			for _, depPkgId := range depPkgIds {
+				// Map the IDs to their names.
+				idToName[depPkgId] = depPkgName
+				idToName[pkgId] = pkgName
+
+				// Skip "guac" files.
+				if dependency.DependencyPackage.Type == "guac" && dependency.DependencyPackage.Namespaces[0].Namespace == "files" {
+					continue
+				}
+
+				// First we need to find all the packages that have pkgName as a dependency
```

Comment:
If I'm reading this correctly, this comment doesn't seem accurate... this bfs finds the dependencies of `depPkgId`, not the dependents of `pkgName`.

Reply:
You are correct, I have changed the comment to `First we need to find all the packages that have pkgName as a dependent`.

Reply:
Sorry to nitpick! But I think this isn't 100% accurate, there may be _other_  `IsDependency` relationships, not yet iterated through and so not accumulated in the two edge sets, that give new direct dependencies of `pkgName`.  This finds the dependencies of `depPkgId` (along with `depPkgId` itself), not `pkgId`/`pkgName`. 

Also, I think the comments on lines 157-163 need similar changes.

Reply:
Thanks, I see what you mean. I have updated the comments to include that we are finding all the packages from the packages that are already scanned.